extension/src/test/suite/experiments/data/index.test.ts

```diff
@@ -177,16 +197,10 @@ suite('Experiments Data Test Suite', () => {
       bypassProcessManagerDebounce(mockNow)
 
       const managedUpdateSpy = spy(data, 'managedUpdate')
-      const dataUpdatedEvent = new Promise(resolve =>
-        data.onDidUpdate(() => resolve(undefined))
-      )
 
-      await Watcher.fireWatcher(mockDotGitFilePath)
-      await dataUpdatedEvent
+      const dataUpdatedEvent = getDataUpdatedEvent(data)
 
-      expect(managedUpdateSpy).to.be.called
-
-      await Watcher.fireWatcher(mockDotGitNestedFilePath)
+      await Watcher.fireWatcher(mockDotGitFilePath)
       await dataUpdatedEvent
 
       expect(managedUpdateSpy).to.be.called
```

Comment:
[F] Because this is the same spy and the same event this part of the test didn't test anything. Can revisit later.